src/java.base/share/classes/java/io/ByteArrayInputStream.java

```diff
@@ -207,10 +207,20 @@ public int readNBytes(byte[] b, int off, int len) {
     public synchronized long transferTo(OutputStream out) throws IOException {
         int len = count - pos;
         if (len > 0) {
+            byte[] tmp;
+            if ("java.io".equals(out.getClass().getPackageName()))
```

Comment:
I think we should trust all classes in `java.*` packages, i.e. the check should be
```java
out.getClass().getPackageName().startsWith("java.")
```

Reply:
Change in 176d5165f7d8f3fa4814c9838abb5d18d9f3c338 to use `java.` instead of `java.io`.